oauth2/src/test/groovy/com/okta/spring/boot/oauth/env/OktaOAuth2PropertiesMappingEnvironmentPostProcessorTest.groovy

```diff
@@ -69,6 +75,31 @@ class OktaOAuth2PropertiesMappingEnvironmentPostProcessorTest {
         assertThat environment.getProperty(TOKEN_URI), nullValue()
         assertThat environment.getProperty(USER_INFO_URI), nullValue()
         assertThat environment.getProperty(PROVIDER_KEYS_URI), nullValue()
+        assertThat environment.getProperty(RS_CLIENT_ID), nullValue()
+        assertThat environment.getProperty(RS_CLIENT_SECRET), nullValue()
+        assertThat environment.getProperty(RS_INTROSPECTION_URI), nullValue()
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    void missingClientSecret() {
+        def environment = buildAndProcessEnvironment([
+            "okta.oauth2.client-id": "test-client-id",
+            "okta.oauth2.issuer": "https://issuer.example.com/foobar",
+            "okta.oauth2.scopes": ["one", "two", "three"],
+        ])
+
+        assertThat environment.getProperty(CLIENT_ID), is("test-client-id")
+        assertThat environment.getProperty(CLIENT_SECRET), nullValue()
+        assertThat environment.getProperty(SCOPE, Set), is(["one", "two", "three"] as Set)
+        assertThat environment.getProperty(ISSUER), is("https://issuer.example.com/foobar")
+        assertThat environment.getProperty(RS_KEYS_URI),is("https://issuer.example.com/foobar/v1/keys")
+        assertThat environment.getProperty(AUTHZ_URI), is("https://issuer.example.com/foobar/v1/authorize")
+        assertThat environment.getProperty(TOKEN_URI), is("https://issuer.example.com/foobar/v1/token")
+        assertThat environment.getProperty(USER_INFO_URI), is("https://issuer.example.com/foobar/v1/userinfo")
+        assertThat environment.getProperty(PROVIDER_KEYS_URI), is("https://issuer.example.com/foobar/v1/keys")
+        assertThat environment.getProperty(RS_CLIENT_ID), is("test-client-id")
+        assertThat environment.getProperty(RS_INTROSPECTION_URI), is("https://issuer.example.com/foobar/v1/introspect")
+        assertThat environment.getProperty(RS_CLIENT_SECRET), nullValue()
```

Comment:
This line should return null instead of throwing an exception

Reply:
noted.